pkg/aws/aws.go

```diff
@@ -26,12 +28,14 @@ const (
 
 //go:generate mockgen -destination mock/stsmock.go -package $GOPACKAGE github.com/aws/aws-sdk-go/service/sts/stsiface STSAPI
 //go:generate mockgen -destination mock/ec2mock.go -package $GOPACKAGE github.com/aws/aws-sdk-go/service/ec2/ec2iface EC2API
+//go:generate mockgen -destination mock/cloudtrailmock.go -package $GOPACKAGE github.com/aws/aws-sdk-go/service/cloudtrail/cloudtrailiface CloudTrailAPI
 
 // Client is a representation of the AWS Client
```

Comment:
Minor nit: // Client breaks the Go Doc strings, because the comment is not starting with AwsClient.
You might want to replace the doc strings with:

// AwsClient is a representation of the Amazon AWS Client.

Reply:
Mhh I just realized that we have a constructor called `NewClient` but we are creating a struct called `AwsClient` with it. I don't know if this is already in use, but it might makes sense to rename one of those for a cleaner documentation.
The Go syntax best practice guide usually suggests that a New<Object> constructor should match the struct name.
So, it should be either NewClient and Client or NewAwsClient and AwsClient:

https://go.dev/doc/effective_go#:~:text=func%20newfile(fd%20int%2C%20name%20string)%20*file%20%7B%20if%20fd%20%3C%200%20%7B%20return%20nil%20%7D%20f%20%3A%3D%20new(file)%20f.fd%20%3D%20fd%20f.name%20%3D%20name%20f.dirinfo%20%3D%20nil%20f.nepipe%20%3D%200%20return%20f%20%7D

That's really just a minor nit and a design decision and this should not block the PR merge process :)